### DISCUSSION TITLE
502 Bad Gateway

### DISCUSSION THREAD

**Discussion by fleeser** [serverpod/serverpod#956] [Q&A] [answered]
I am trying to get the server up and running on AWS for two days now, but get a **502 Bad Gateway** when trying to access `api.app.example.app`. 

The terraform scripts seem to work since they run successfully. 

**Background Info**:
- I am using a subdomain `app.example.app` as the base domain. The generated subdomains then would be e.g. `api.app.example.app`.
- I am using the `eu-central-1` region besides the required `us-east-1` region.

I am able to connect to the database via tools like pgAdmin. When trying to access all the other subdomains via the Browser, the **502 Bad Gateway** response code gets returned.

I also found this discussion: [Discussion](https://github.com/serverpod/serverpod/discussions/445), but I don't think that a wrong Dart version is causing this error since I already replaced all the SDK versions in all files by nearly every possible version out there.

I tried creating a complete new Serverpod project, so that the Dart SDK versions are the default ones and the same error appears.

**Comment by sroepges** [comment]
Same here!

**Reply by sroepges** [reply, marked as answer]
I came up with a solution, my solution focussed on the deployment to AWS.
Before I could even deploy to AWS I had to setup some things.

Everytime I an the command `terraform apply`, I got an error saying that my S3 buckets don't support ACLs. There was also another issue that someone else fixed: #925.
With this changed, I was able to deploy to AWS.

After deploying to AWS and setting up the database I ran the GitHub action `deployment-aws.yml`.
When I opened `api.example.com` I got the response **502 Bad Gateway**.
I checked the EC2 instance and saw that no code got deployed from AWS CodeDeploy, so I checked the scripts that were executed on the EC2. The script `init-script.sh` throws an error in line 24 (`no permission`), so I changed it to be `sudo ./install auto`. Now when I start the GitHub action again the CodeDeploy agent is installed and the server files are served to the correct location `serverpod/active/`.

Additionally I had to change the **database** and **redis** hosts of the `production.yaml` & `staging.yaml`.
In `production.yaml` I changed `host: database.private-production.examplepod.com` to `host: database.examplepod.com` and `host: redis.private-production.examplepod.com` to `host: redis.examplepod.com`.
In `staging.yaml` I changed `host: database.private-staging.examplepod.com` to `host: database-staging.examplepod.com` and `host: redis.private-staging.examplepod.com` to `host: redis-staging.examplepod.com`.

With all the above steps completed you can trigger the GitHub workflow again and after it's finished you should get no **502 Bad Gateway** response.

**Reply by fleeser** [reply]
That solved the error! Thank you!

**Comment by fleeser** [comment]
Solution found in comment from @sroepges
